redis/_parsers/resp3.py

```diff
@@ -225,12 +226,16 @@ async def _read_response(
                 pass
         # map response
         elif byte == b"%":
-            response = {
-                (await self._read_response(disable_decoding=disable_decoding)): (
-                    await self._read_response(disable_decoding=disable_decoding)
+            # we use this approach and not dict comprehension here
+            # because this dict comprehension fails in python 3.7
+            # (evaluation order of key:val is _undefined_ in Python)
```

Comment:
Do you know if it's _undefined_ also in python 3.8+?

Reply:
I'm not sure that we need to fix it because python 3.7 is EOL and we are going to drop support on it in 5.1

Reply:
but it is still supported, though.  as it is, 3.7 will fail parsing resp3.  There is already a fix for the "sync" version, why not fix the async one?

Reply:
Not pushing for it, though.  just putting in an argument :).   For my other PR where this came up, I can certainly put in a skip for 3.7